packages/xo-server-sdn-controller/src/index.js

```diff
@@ -168,38 +168,40 @@ class SDNController extends EventEmitter {
     // FIXME: we should monitor when xapis are added/removed
     forOwn(this._xo.getAllXapis(), async xapi => {
       await xapi.objectsFetched
+      if (this._setControllerNeeded(xapi)) {
+        return
+      }
 
-      if (this._setControllerNeeded(xapi) === false) {
-        this._cleaners.push(await this._manageXapi(xapi))
+      this._cleaners.push(await this._manageXapi(xapi))
+      const hosts = filter(xapi.objects.all, { $type: 'host' })
+      await Promise.all(
```

Comment:
Does all of this need to be done *before* the plugin can be marked as "loaded"? If so, `_.forOwn` won't help you and you'll either need a higher level `Promise.all` or a `for` loop if you need to do it sequentially.

Reply:
Yes, it is the plugin initialisation.